src/main/java/ru/l1ttleO/boyfriend/settings/Settings.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    protected <K> void loadEntry(final K key, final @NotNull Entry<K, ?> entry) {
+        try {
+            if (this.hasProperty(entry.name))
+                entry.putFromString(key, this.getProperty(entry.name));
+        } catch (final IllegalArgumentException e) {
+            System.err.println(e);
+        }
+    }
+
+    public static abstract class Entry<K, V> {
+        protected final Map<K, V> VALUES = new HashMap<>(); // what is storing actual values
+        protected final Function<K, Settings> settingsGetter;
+        public final String name;
+        public final String type;
+        public final V defaultValue;
+
+        public Entry(final @NotNull Function<K, Settings> settingsGetter, final @NotNull String name, final @NotNull String type, final V defaultValue) {
+            this.settingsGetter = settingsGetter;
+            this.name = name;
+            this.type = type;
+            this.defaultValue = defaultValue;
+        }
+
+        protected boolean save(final K key) {
+            return this.settingsGetter.apply(key).save(key, this);
+        }
+
+        public boolean has(final K key) {
+            this.settingsGetter.apply(key); // make sure settings are initialized and loaded
+            return this.VALUES.containsKey(key);
+        }
+
+        public V get(final K key) {
+            this.settingsGetter.apply(key); // make sure settings are initialized and loaded
+            return this.VALUES.getOrDefault(key, this.defaultValue);
+        }
+
+        protected void put(final K key, final V value) {
+            this.VALUES.put(key, value);
+        }
+
+        public void set(final K key, final V value) {
+            this.put(key, value);
+            this.save(key);
+        }
+
+        public void reset(final K key) {
+            this.VALUES.remove(key);
+            this.save(key);
+        }
+
+        public @Nullable String getString(final K key) {
+            final V value = this.get(key);
+            return value == null ? null : value.toString();
+        }
+
+        public abstract V putFromString(final K key, final @NotNull String str);
+
+        public @Nullable String formatted(final K key) {
+            final String str = this.getString(key);
+            return str == null ? null : "`" + str + "`";
+        }
+
+        public V setFormatted(final K key, final String str) throws IllegalArgumentException {
+            final V value = this.putFromString(key, str);
+            this.save(key);
+            return value;
+        }
+    }
+
+    public static class StringEntry<K> extends Entry<K, String> {
+        public StringEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name, final String defaultValue) {
+            super(keyGetter, name, "string", defaultValue);
+        }
+
+        @Override
+        public String putFromString(final K key, final String str) throws IllegalArgumentException {
```

Comment:
```suggestion
        public String putFromString(final K key, final @NotNull String str) throws IllegalArgumentException {
```
`Overriden method parameters are not annotated` [intellij] [1/4]